Client/WebAuthN/U2FExtensions.swift

```diff
@@ -121,83 +129,157 @@ class U2FExtensions: NSObject {
+                        self?.observeSessionStateUpdates = false
+                        self?.handleSessionStateChange()
+                    }
+                })
+
+                nfcSessionStateObservation = nfcSession.observe(\.iso7816SessionState, changeHandler: { [weak self] session, change in
+                    ensureMainThread {
+                        self?.observeSessionStateUpdates = false
+                        self?.handleSessionStateChange()
+                    }
+                })
             } else {
-                keySession.removeObserver(self, forKeyPath: #keyPath(YKFKeySession.sessionState))
+                accessorySessionStateObservation = nil
+                nfcSesionStateObservation = nil
             }
         }
     }
     
-    private var observeKeyStateUpdates: Bool = false {
+    private var observeServiceStateUpdates: Bool = false {
         didSet {
-            if oldValue == observeKeyStateUpdates {
+            if oldValue == observeServiceStateUpdates {
                 return
             }
             
-            let keySession = YubiKitManager.shared.keySession as AnyObject
+            let keySession = YubiKitManager.shared.accessorySession as AnyObject
             
-            if observeKeyStateUpdates {
-                keySession.addObserver(self, forKeyPath: #keyPath(YKFKeySession.fido2Service.keyState), options: [], context: &U2FExtensions.observationContext)
-                keySession.addObserver(self, forKeyPath: #keyPath(YKFKeySession.u2fService.keyState), options: [], context: &U2FExtensions.observationContext)
+            if observeServiceStateUpdates {
+                keySession.addObserver(self, forKeyPath: #keyPath(YKFAccessorySession.fido2Service.keyState), options: [], context: &U2FExtensions.observationContext)
+                keySession.addObserver(self, forKeyPath: #keyPath(YKFAccessorySession.u2fService.keyState), options: [], context: &U2FExtensions.observationContext)
             } else {
-                keySession.removeObserver(self, forKeyPath: #keyPath(YKFKeySession.fido2Service.keyState))
-                keySession.removeObserver(self, forKeyPath: #keyPath(YKFKeySession.u2fService.keyState))
+                keySession.removeObserver(self, forKeyPath: #keyPath(YKFAccessorySession.fido2Service.keyState))
+                keySession.removeObserver(self, forKeyPath: #keyPath(YKFAccessorySession.u2fService.keyState))
             }
         }
     }
+    
+    private var keyType: FIDO2ViewControllerKeyType = .unknown
+    
+    /**
+     Returns the service associated with the desired session.
+     */
+    private var keyFido2Service: YKFKeyFIDO2ServiceProtocol? {
+        get {
+            if keyType == .accessory {
+                return YubiKitManager.shared.accessorySession.fido2Service
+            } else {
+                if #available(iOS 13.0, *) {
+                    return YubiKitManager.shared.nfcSession.fido2Service
```

Comment:
Hmm.. is there no backwards compatibility for when it's not just an accessory?
iOS 11+ supports CoreNFC as well.

Reply:
Read/Write capability for NFC was supported in iOS 13, so the YubiKit APIs are gated by this check. For iOS 11/12, we just present the option to use lightning keys.